monkey/infection_monkey/network/mssql_fingerprinter.py

```diff
@@ -0,0 +1,102 @@
+import errno
+import logging
+import socket
+from typing import Any, Dict, Optional
+
+from infection_monkey.i_puppet import FingerprintData, IFingerprinter, PingScanData, PortScanData
+
+MSSQL_SERVICE = "MSSQL"
+SQL_BROWSER_DEFAULT_PORT = 1434
+_BUFFER_SIZE = 4096
+_MSSQL_SOCKET_TIMEOUT = 5
+
+logger = logging.getLogger(__name__)
+
+
+class MSSQLFingerprinter(IFingerprinter):
+    def get_host_fingerprint(
+        self,
+        host: str,
+        _: PingScanData,
+        port_scan_data: Dict[int, PortScanData],
+        options: Dict,
+    ):
+        """Gets Microsoft SQL Server instance information by querying the SQL Browser service."""
+        services = {}
+
+        try:
+            # Create a UDP socket and sets a timeout
+            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            sock.settimeout(_MSSQL_SOCKET_TIMEOUT)
```

Comment:
This can be done in `_receive_socket_data()`, since nothing else uses this socket.